### DISCUSSION TITLE
Does simulateContract estimate the contract call gas limit?

### DISCUSSION THREAD

**Discussion by Sam-Scolari** [wevm/viem#862] [Question] [answered]
I ran into this issue today trying to send a transaction

```
Insufficient funds for gas * price + value: address 0x035342Fb880F46A9F58343774F131Bf6f6757007 have 77717615752227236 want 13418145387500000000 (supplied gas 500000000)

Contract Call:
  maxPriorityFeePerGas:  8.945430258 gwei
  maxFeePerGas:          26.836290775 gwei
```

The supplied gas is really high making it think I want to send over 13 ETH worth of gas for the transaction. I couldn't find any reference to a gasLimit parameter in the docs but noticed a parameter called gas in my IDE (not specified in the docs). Is this the gas limit and should I call estimateContractGas myself and pass it into the simulateContract?

**Comment by jxom** [comment, marked as answer]
Yep! `gas` is the gas limit, and you should estimate gas using estimateContractGas. 

Are you using a JSON-RPC account? If so, I wonder why the wallet is calculating the gas so high for you.

**Reply by Sam-Scolari** [reply]
I set the maxFeePerGas intentionally high for MEV purposes but I wasn't using estimateContractGas so it was defaulting to 500,000,000 as the gas limit I guess. Works fine now that I am using estimateContractGas, thanks.

**Comment by kasparkallas** [comment]
Can gas estimation and contract simulation be combined into a single viem call? Doesn't estimating the gas effectively already simulate the contract call?

**Reply by hungdoansy** [reply]
Had same thought
